tensorflow/compiler/xla/service/hlo_instructions.h

```diff
@@ -1257,6 +1261,8 @@ class HloCustomCallInstruction : public HloInstruction {
   // For layout-constrained custom calls, this vector holds the shape with
   // layout for each operand.
   std::vector<Shape> operand_shapes_with_layout_;
+  // Whether this call has a side-effect - by default there are no side-effects.
```

Comment:
Perhaps the comment about the default would be more useful on set_has_side_effect?  (Or we could have it in both places.)